eeps/eep-0072.md

```diff
@@ -0,0 +1,195 @@
+information about the error that would not be practical to present
+directly to the user. Here is an example of what the entry could look
+like for the above code:
+
+![Erlang Error Index Sample Entry][]
+
+Unique error codes also have the advantage to be better searchable in
+forums and chats, where the exact error message could vary, but the
+error code would be the same.
+
+Finally, error codes can be used by IDEs (e.g. via language servers)
+to match on error codes and provide contextual help. Both the [Erlang
+LS][] and the [ELP][] language server already use "unofficial" error
+codes.
+
+Format
+------
+
+An error code should be composed by two parts: an alphanumeric
+_namespace_ (three letters) and a numeric identifier (four digits),
+divided by a dash (`-`).
+
+A potential set of namespaces could look like the following:
+
+| Namespace | Description                                                     |
+|-----------|-----------------------------------------------------------------|
+| ERL       | The Erlang compiler and related tools (linter, parser, scanner) |
+| DIA       | The Dialyzer type-checker                                       |
+| ELV       | The Elvis code-style reviewer                                   |
+| ELP       | The Erlang Language Platform                                    |
+| ...       | ...                                                             |
+
+A set of potential error codes could look like:
+
+    ERL-0123
+    DIA-0009
+    ELV-0015
+    ELP-0001
+
+The exact number of characters/digits for each namespace and code is
+open for discussion, as well as the fact whether components such as
+the parser, the scanner or the `erlint` Erlang linter should have
+their own namespace.
+
+Responsibilities
+----------------
+
+The Erlang/OTP team would be ultimately responsible for maintaining a
+  list of _official_ namespaces. Each tool maintainer would then be
+  responsible to allocate specific codes to specific diagnostics.
+
+Processes
+---------
+
+Error cannot be re-used. If a tool stops emitting an error code, the
+_deprecated_ error code is still documented in the index, together
+with a deprecation notice. This is to avoid re-using a single code for
+multiple purposes.
+
+Reference Implementation
+------------------------
+
+The [ELP website][] contains a proof of concept of what an Erlang
+Error Index could look like. Ideally, such a website would live under
+the `erlang.org` domain.
+
+The website should use _Markdown_ as the primary mechanism to write
+content and it should be easily extensible by the Community.
+
+Required Changes
+================
+
+* The `erlc` compiler and the `dialyzer` type-checker would start
+  emitting error codes together with errors and warnings messags.
+* The tools could also expose a `--explain` option from the command
+  line - similarly to `rustc` - which would take an error code as the
+  input and return the Markdown documentation as output.
+
```

Comment:
Would it make sense to extend the proposal to include a machine-readable format (JSON?) for reporting errors

Reply:
IMO that is a separate issue. This EEP should focus on the index and how to allocate names/codes to it.